src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -891,11 +908,41 @@ private AttributeListSyntax ParseAttributeDeclaration()
 
             var attributes = _pool.AllocateSeparated<AttributeSyntax>();
             this.ParseAttributes(attributes);
-            return _syntaxFactory.AttributeList(
-                openBracket,
-                location,
-                _pool.ToListAndFree(attributes),
-                this.EatToken(SyntaxKind.CloseBracketToken));
+
+            var closeBracket = this.EatToken(SyntaxKind.CloseBracketToken);
+            if (inExpressionContext && parseAsCollectionLiteral())
+            {
+                // we're in an expression and we've seen `[A, B].`  This is actually the start of a collection literal
+                // that someone is explicitly accessing a member off of.
+                _pool.Free(attributes);
+                resetPoint.Reset();
+                return null;
+            }
+
+            return _syntaxFactory.AttributeList(openBracket, location, _pool.ToListAndFree(attributes), closeBracket);
+
+            bool parseAsCollectionLiteral()
```

Comment:
This doesn't actually parse anything. Perhaps a better name is `shouldParseAsCollectionLiteral`?

Reply:
done.